### DISCUSSION TITLE
about timeline

### DISCUSSION THREAD

**Discussion by paprophet** [iTwin/itwinjs-core#4049] [Q&A] [answered]
now i want show the middle part of the element on first time ,and then show all part of the element on end time.
how to achieve this effect by the vp.displayStyle.settings.scheduleScriptProps?

**Comment by pmconne** [comment]
The script's timeline operates on elements, not on portions of elements. The only way it can control the visibility of part of an element is by applying a [clip plane](https://www.itwinjs.org/reference/core-common/displaystyles/renderschedule/renderschedule.cuttingplaneentry/) - but that is only one plane, so while it can control the visibility of one end or another of the element, it can't hide both ends while showing the middle part.

If your element was defined as 3 elements (e.g., left, middle, right), optionally with an assembly parent, then you could control the visibility of each part individually.

**Reply by paprophet** [reply]
ok
can  the cuttingPlaneTimeline support different value of same element at the same time,superpose the different value,
for example:
cuttingPlaneTimeline = [
                     **{
                        value: {
                            position: position,
                            direction: direction1,
                            visible: false,
                            hidden: false,
                        },
                        time:startTime ,
                        interpolation: 2,
                    },
                     {
                        value: {
                            position: position,
                            direction: direction2,
                            visible: false,
                            hidden: false,
                        },
                        time:startTime ,
                        interpolation: 2,
                    },** 
                    {
                        value: {
                            position: position,
                            direction: direction_end,
                            visible: true,
                            hidden: true,
                        },
                        time:  finishTime ,
                        interpolation: 2,
                    },
                ];

**Reply by pmconne** [reply]
It currently supports only a single cutting plane. We can consider adding support for multiple cutting planes, but the rules for interpolating between them - especially if, as in your example, you change the number of cutting planes halfway through the timeline - will need to be established.

**Reply by paprophet** [reply]
ok.if the Timeline have **cuttingBoxTimeline**  to set script is perfect.

**Comment by pmconne** [comment, marked as answer]
Please follow #4063 for updates. Currently unscheduled.
